Find a functionally equivalent code:android.text.PackedIntVector.growBuffer:COMMENT
Method Modifier: private     final       
Comment:/**
 * Grows the value and gap arrays to be large enough to store at least
 * one more than the current number of rows.
 */

@@ -1,8 +1,7 @@
 {
     final int columns = mColumns;
-    int newsize = size() + 1;
-    newsize = ArrayUtils.idealIntArraySize(newsize * columns) / columns;
-    int[] newvalues = new int[newsize * columns];
+    int[] newvalues = ArrayUtils.newUnpaddedIntArray(GrowingArrayUtils.growSize(size()) * columns);
+    int newsize = newvalues.length / columns;
     final int[] valuegap = mValueGap;
     final int rowgapstart = mRowGapStart;
     int after = mRows - (rowgapstart + mRowGapLength);

